vectorizer.py

```diff
@@ -1,9 +1,67 @@
 from sklearn.feature_extraction.text import TfidfVectorizer
+import json, warnings
+warnings.simplefilter("ignore", UserWarning)
 
 class Vectorizer(TfidfVectorizer):
-    def retrain(self, fp):
-        data = []
-        with open(fp, 'rb') as fl:
-            for i, line in enumerate(fl):
-                data.append(line)
-        self.vocabulary_ = dict(self.vocabulary_, **self.fit(data).vocabulary_)
\ No newline at end of file
+    def train(self, fp, samples:int=None, seeker:int=None, debug:bool=False):

```

Comment:
samples:int=None causes a problem downstream if a number of samples is not provided.